### DISCUSSION TITLE
Server Actions in Next14

### DISCUSSION THREAD

**Discussion by Jauki** [vercel/next.js#57714] [Help]
### Summary

Hello everyone,

I'm currently facing an issue after a recent upgrade. The problem is as follows: I have a Dialog that serves as an Input Form, and my goal is to send this formData to a server action. However, since the upgrade, this functionality has stopped working, and I'm encountering the following error:

```
react_dom__WEBPACK_IMPORTED_MODULE_6__.useFormState is not a function
```

In my efforts to debug this issue, I've been using useFormState and useFormStatus to retrieve error codes.

Additionally, I made adjustments to accommodate the new server actions. I was unsure if this had changed because in the past, I used prevState, but I no longer have access to it in my server action. To address this, I moved the code from the action.ts file into the Form Component. I've also included a "useServer," but I'm not receiving any output in the console.

Here's my createRoom server action:

```ts
export async function createRoom(formData: FormData) {
  "use server";
  // Authorization Stuff
  const session = await getServerSession(authOptions);
  if (!session) {
    return { message: 'Unauthorized' };
  }

  try {
    // Perform necessary actions
    return revalidatePath('/');
  } catch (e) {
    console.error(e);
    return { message: 'Failed to create' };
  }
}
```
On the client side, I've defined the form as follows:

```jsx
<form action={createRoom}>
...
</form>
```

I'm seeking assistance to resolve this issue. Any help or insights would be greatly appreciated.

### Additional information

```js
Operating System:
  Platform: darwin
  Arch: x64
  Version: Darwin Kernel Version 21.6.0: Thu Sep 29 20:12:57 PDT 2022; root:xnu-8020.240.7~1/RELEASE_X86_64
Binaries:
  Node: 18.17.0
  npm: 10.2.0
  Yarn: 1.22.19
  pnpm: N/A
Relevant Packages:
  next: 14.0.0
  eslint-config-next: 13.5.4
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.6
Next.js Config:
  output: N/A
```

**Comment by devraj** [comment]
I ran into something similar and I think the error (at least for me) was that it's recommended to have the form in file with the `'use client'` annotation and the `action` in a separate file with `'use server'` annotation and then use the `useFormState` hook which is in `react-canary` but `next14` builds agains that.

See [this example](https://github.com/vercel/next.js/blob/canary/examples/next-forms/app/add-form.tsx) repository

**Reply by Jauki** [reply]
Actually what fixed my problem was this: https://github.com/vercel/next.js/issues/56041#issuecomment-1771416368
